ml-agents/mlagents/trainers/trainer/rl_trainer.py

```diff
@@ -122,6 +124,23 @@ def create_tf_policy(
         """
         pass
 
+    def create_saver(self, policy: Policy) -> BaseSaver:
+        if self.framework == "torch":
+            saver = TorchSaver(  # type: ignore
+                policy,  # type: ignore
```

Comment:
Question : How is the Optimizer passed to the Saver?

Reply:
See `add_policy()` in ppo/trainer.py or sac/trainer.py of the actual usage. The complete flow be like:
1. When adding the first policy, a saver is created and associated with that policy. Main purpose to have to policy here is to reset global steps.
2. call `saver.register(policy)`. It will get modules by calling policy.get_modules() and store them to self.modules
3. call `saver.register(optimizer)`. It will get modules by calling optimizer.get_modules() and store them to self.modules
4. When saving/loading, the saver iterate through self.modules and save/load by `state_dict()`/`load_state_dict()`.